### PR TITLE
Bug 1576728 - Fixes multiple requests to a CR

### PR DESCRIPTION
**Pull request by shawn-hurley**
#### Describe what this PR does and why we need it:
This makes the dao calls happen 1 at a time for a dao. Attempts to handle the case when two service instances are gotten with two bindings. both are deleted, but because we only remove each of one, and when we update we just set the value, we were adding back a binding.

Depends on https://github.com/automationbroker/bundle-lib/pull/80

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -44,6 +44,7 @@
   version = "0.1.2"
 
 [[projects]]
+  branch = "bug-1576728"
```
Comment (jmrodri) [inline Gopkg.lock]: will this remain when we merge?

File: Gopkg.toml [hunk 2]
```diff
@@ -37,7 +37,9 @@
   version = "~1.3.0"
 
 [[constraint]]
-  version = "~0.1.0"
+  #version = "~0.1.0"
+  branch = "bug-1576728"
+  source = "github.com/shawn-hurley/bundle-lib"
```
Comment (jmrodri) [inline Gopkg.toml]: are we doing this?

File: pkg/dao/crd/dao.go [hunk 3]
```diff
@@ -47,13 +48,20 @@ const (
...
 }
 
 // NewDao - Create a new Dao object
 func NewDao(namespace string) (*Dao, error) {
-	dao := Dao{namespace: namespace}
+	dao := Dao{namespace: namespace,
+		bundleLock:   sync.Mutex{},
+		bindingLock:  sync.Mutex{},
+		instanceLock: sync.Mutex{},
+	}
```
Comment (jmrodri) [inline pkg/dao/crd/dao.go]: CRDs suck as a database :)

File: Gopkg.toml (hunk 2 above)
Comment (shawn-hurley) [reply Gopkg.toml]: No, this depends on this https://github.com/automationbroker/bundle-lib/pull/80

File: pkg/dao/crd/dao.go [hunk 4]
```diff
@@ -47,13 +48,20 @@ const (
 
 // Dao - object to interface with the data store.
 type Dao struct {
-	client    automationbrokerv1.AutomationbrokerV1alpha1Interface
-	namespace string
+	client       automationbrokerv1.AutomationbrokerV1alpha1Interface
+	namespace    string
+	bundleLock   sync.Mutex
```
Comment (eriknelson) [inline pkg/dao/crd/dao.go]: :+1:

**Comment by rthallisey** [comment]
https://github.com/automationbroker/bundle-lib/pull/80 just merged. Should be able to remove  ```branch = "bug-1576728"``` everywhere

**Comment by eriknelson** [comment]
@rthallisey I think we probably want to update the Gopkg stuff to reference a release tag of bundle-lib.

**Comment by eriknelson** [comment]
This is ready to merge with the bundle lib release.

**Pull request merged**